baremetal/metal3machine_manager.go

```diff
@@ -552,6 +580,67 @@ func (m *MachineManager) Delete(ctx context.Context) error {
 			return &RequeueAfterError{RequeueAfter: requeueAfter}
 		}
 
+		// Fetch corresponding Metal3MachineTemplate, to see if NodeReuse
+		// feature is enabled. If set to true, check the machine role. In case
+		// machine role is ControlPlane, set nodeReuseLabelName to KubeadmControlPlane
+		// name, otherwise to MachineDeployment name.
+		m.Log.Info("Getting Metal3MachineTemplate")
+		m3mt := &capm3.Metal3MachineTemplate{}
+		if m.Metal3Machine == nil {
+			return errors.New("Metal3Machine associated with Metal3MachineTemplate is not found")
+		}
+		if m.hasTemplateAnnotation() {
+			m3mtKey := client.ObjectKey{
+				Name:      m.Metal3Machine.ObjectMeta.GetAnnotations()[capi.TemplateClonedFromNameAnnotation],
+				Namespace: m.Metal3Machine.Namespace,
+			}
+			if err := m.client.Get(ctx, m3mtKey, m3mt); err != nil {
+				// we are here, because while normal deprovisioning, Metal3MachineTemplate will be deleted first
+				// and we can't get it even though Metal3Machine has reference to it. We consider it nil and move
+				// forward with normal deprovisioning.
+				m3mt = nil
+				m.Log.Info("Metal3MachineTemplate associated with Metal3Machine is deleted")
+			} else {
+				// in case of upgrading, Metal3MachineTemplate is still and not deleted there and we can get it,
```

Comment:
`still not deleted` may be is the correct phrase

Reply:
Fixed, thanks